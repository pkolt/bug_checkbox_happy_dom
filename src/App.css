body {
  height: 100vh;
}

#app {
  display: flex;
  flex-direction: column;
  margin: 0 25%;
}

.field {
    display: flex;
    align-items: center;
    gap: 0.2rem;
}