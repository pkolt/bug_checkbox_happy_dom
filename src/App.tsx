import { useState } from 'react';
import './App.css';

const App = () => {
  const [coffee, setCoffee] = useState(false);
  const [tea, setTea] = useState(false);
  return (
    <div id="app">
      <form>
        <fieldset>
          <legend>Menu</legend>
          <p className="field">
            <input
              type="checkbox"
              name="coffee"
              id="id_coffee"
              onChange={() => setCoffee((state) => !state)}
              checked={coffee}
              data-testid="coffee"
            />
            <label htmlFor="id_coffee">Coffee</label>
          </p>
          <p className="field">
            <input
              type="checkbox"
              name="tea"
              id="id_tea"
              onChange={() => setTea((state) => !state)}
              checked={tea}
              data-testid="tea"
            />
            <label htmlFor="id_tea">Tea</label>
          </p>
        </fieldset>
      </form>

      <p>
        {coffee && (
          <>
            Selected Coffee
            <br />
          </>
        )}
        {tea && (
          <>
            Selected Tea
            <br />
          </>
        )}
      </p>
    </div>
  );
};

export default App;
