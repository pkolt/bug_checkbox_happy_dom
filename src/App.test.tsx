import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('App', () => {
  it('render', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('select Tea', async () => {
    const { container } = render(<App />);
    const checkbox = screen.getByTestId('tea');
    await userEvent.click(checkbox);
    screen.debug(container);
    expect(container.innerHTML).toMatch('Selected Tea');
  });
});
