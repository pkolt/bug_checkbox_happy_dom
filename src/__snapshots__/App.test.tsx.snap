// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > render 1`] = `
<div>
  <div
    id="app"
  >
    <form>
      <fieldset>
        <legend>
          Menu
        </legend>
        <p
          class="field"
        >
          <input
            data-testid="coffee"
            id="id_coffee"
            name="coffee"
            type="checkbox"
          />
          <label
            for="id_coffee"
          >
            Coffee
          </label>
        </p>
        <p
          class="field"
        >
          <input
            data-testid="tea"
            id="id_tea"
            name="tea"
            type="checkbox"
          />
          <label
            for="id_tea"
          >
            Tea
          </label>
        </p>
      </fieldset>
    </form>
    <p />
  </div>
</div>
`;
